apps/rush-lib/src/cli/RushPnpmCommandLine.ts

```diff
@@ -0,0 +1,259 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import colors from 'colors/safe';
+import { FileSystem, AlreadyReportedError, Executable, EnvironmentMap } from '@rushstack/node-core-library';
+import { PrintUtilities } from '@rushstack/terminal';
+
+import { RushConfiguration } from '../api/RushConfiguration';
+import { NodeJsCompatibility } from '../logic/NodeJsCompatibility';
+import { SpawnSyncReturns } from 'child_process';
+
+export interface ILaunchRushPnpmInternalOptions {
+  isManaged: boolean;
+  alreadyReportedNodeTooNewError?: boolean;
+}
+
+const RUSH_SKIP_CHECKS_PARAMETER: string = '--rush-skip-checks';
+
+enum CommandKind {
+  KnownSafe,
+  ShowWarning,
+  Blocked
+}
+
+export class RushPnpmCommandLine {
+  private static _knownCommandMap: Map<string, CommandKind> = new Map<string, CommandKind>([
+    ['add', CommandKind.ShowWarning],
+    ['audit', CommandKind.KnownSafe],
+    ['exec', CommandKind.KnownSafe],
+    ['import', CommandKind.Blocked],
+    ['install', CommandKind.Blocked],
+    [/* synonym */ 'i', CommandKind.Blocked],
+    ['install-test', CommandKind.Blocked],
+    [/* synonym */ 'it', CommandKind.Blocked],
+    ['link', CommandKind.ShowWarning],
+    [/* synonym */ 'ln', CommandKind.ShowWarning],
+    ['list', CommandKind.KnownSafe],
+    [/* synonym */ 'ls', CommandKind.KnownSafe],
+    ['outdated', CommandKind.KnownSafe],
+    ['pack', CommandKind.KnownSafe],
+    ['prune', CommandKind.KnownSafe],
+    ['publish', CommandKind.KnownSafe],
+    ['rebuild', CommandKind.KnownSafe],
+    [/* synonym */ 'rb', CommandKind.KnownSafe],
+    ['remove', CommandKind.ShowWarning],
+    [/* synonym */ 'rm', CommandKind.ShowWarning],
+    ['root', CommandKind.KnownSafe],
+    ['run', CommandKind.KnownSafe],
+    ['start', CommandKind.KnownSafe],
+    ['store', CommandKind.KnownSafe],
+    ['test', CommandKind.KnownSafe],
+    [/* synonym */ 't', CommandKind.KnownSafe],
+    ['unlink', CommandKind.ShowWarning],
+    ['update', CommandKind.ShowWarning],
+    [/* synonym */ 'up', CommandKind.ShowWarning]
```

Comment:
```suggestion
    [/* synonym */ 'up', CommandKind.ShowWarning],
    ['why', CommandKind.KnownSafe]
```

Looks like `why` command is missing.  Note that with useWorkspaces on, `rush-pnpm why something` will always print nothing, but `rush-pnpm why -r something` is useable and pretty useful.